tenable/io/v3/vm/agents/api.py

```diff
@@ -0,0 +1,124 @@
+'''
+Agents
+======
+
+The following methods allow for interaction into the Tenable.io
+:devportal:`agents <agents>` API endpoints.
+
+Methods available on ``tio.v3.vm.agents``:
+
+.. rst-class:: hide-signature
+.. autoclass:: AgentsAPI
+    :members:
+'''
+from typing import Dict, Union
+from uuid import UUID
+
+from tenable.io.v3.base.endpoints.explore import ExploreBaseEndpoint
+from tenable.io.v3.vm.agents.schema import AgentSchema
+
+
+class AgentsAPI(ExploreBaseEndpoint):
+    '''
+    This class contains methods related to Agents API
+    '''
+    _path: str = 'api/v3/agents'
+    _conv_json: bool = True
+    _schema = AgentSchema()
+
+    def details(self, agent_id: UUID) -> Dict:
+        '''
+        Retrieves the details of an agent.
+
+        :devportal:`agents: get <agents-get>`
+
+        Args:
+            agent_id (UUID): The unique identifier of the agent.
+
+        Returns:
+            :obj:`dict`:
+                The agent dictionary record.
+
+        Examples:
+            >>> agent = tio.v3.vm.agents.details(
+            ...     '00000000-0000-0000-0000-000000000000'
+            ... )
+            >>> pprint(agent)
+        '''
+        return self._get(f'{agent_id}')
+
+    def list_agents_from_group(self):
+        raise NotImplementedError(
+            "This method will be implemented later."
+        )
+
+    def task_status(self, task_uuid: UUID) -> Dict:
+        '''
+        Retrieves the current status of the task requested.
+
+        :devportal:`bulk-operations: bulk-agent-status
+        <bulk-task-agent-status>`
+
+        Args:
+            task_uuid (UUID): The unique identifier of the agent.
+
+        Returns:
+            :obj:`dict`:
+                Task resource
+
+        Examples:
+            >>> item = tio.v3.vm.agents.unlink(
+            ...     '00000000-0000-0000-0000-000000000000',
+            ...     '00000000-0000-0000-0000-000000000001',
+            ...     '00000000-0000-0000-0000-000000000002'
+            ... )
+            >>> task = tio.v3.vm.agent.task_status(item['task_id'])
+            >>> pprint(task)
+        '''
+        return self._get(f'_bulk/{task_uuid}')
+
+    def search(self):
+        raise NotImplementedError('This method will be implemented later.')
+
+    def unlink(self, *agent_ids: UUID) -> Union[Dict, None]:
```

Comment:
Should use the DELETE action like we have here: https://developer.tenable.com/reference#agents-delete

Reply:
In this method, we are using the DELETE action for singular agent_id as mentioned in API documentation but for multiple agent_id we are using POST action which is covered under the agent bulk operation. So, basically, this will enable end-user to delete single or multiple agent IDs using the same function.
Ref: https://developer.tenable.com/reference#bulk-unlink-agents

Reply:
Ok, got it